CRM/Contribute/Form/Contribution.php

```diff
@@ -209,13 +209,26 @@ class CRM_Contribute_Form_Contribution extends CRM_Contribute_Form_AbstractEditP
    */
   public $payment_instrument_id;
 
+  /**
+   * @var bool
+   */
+  private $_payNow;
+
   /**
    * Explicitly declare the form context.
    */
   public function getDefaultContext() {
     return 'create';
   }
 
+  public function __get($name) {
```

Comment:
I like this way of throwing the deprecation warning, but just noting this could cause issues if anyone is doing `property_exists($form, '_contributionID');`, as `property_exists` won't run `__get()`.

Difficult to judge how likely that is. If we wanted to fix it we could declare `private $_contributionID;` without giving it a value. This would make `property_exists` return true, but the `__get()` would still be called as the property would be private.

Reply:
@braders yeah good point - I think it's possible to implement `__isset` for that - but for undeclared properties it would have already returned FALSE I think?